### PR TITLE
[sailfish-browser] Limit history size to 2000 entries

### LINKED ISSUE #199: Remove oldest browsing history entries from database
Oldest entries should be periodically cleaned from browser's history database. Either keep only X amount of latest entries or timestamp entries and remove entries older than X days.

2-3months time should be enough, while it can still be a lot.

As history database queries are executed in another thread, clean-up could be for instance done when browser is started.

### PR DESCRIPTION
**Pull request by siteshwar**
Fixes #199

### CODE REVIEW

File: src/dbworker.cpp [hunk 1]
```diff
@@ -107,6 +107,13 @@ void DBWorker::init()
             QSqlQuery query = prepare(db_schema[i]);
             execute(query);
         }
+    } else {
+        // Limit history size to 2000 entries
+        QSqlQuery cleanupHistory = prepare("DELETE FROM browser_history WHERE id NOT IN (SELECT id from "\
+                                           "browser_history ORDER BY date DESC LIMIT 2000);");
```
Comment (rojkov) [inline src/dbworker.cpp]: I'd better define the number in a macros closer to the top of the file

Comment (siteshwar) [inline src/dbworker.cpp]: Fixed through 2b83f0f

File: src/dbworker.cpp [hunk 2]
```diff
@@ -107,6 +109,13 @@ void DBWorker::init()
             QSqlQuery query = prepare(db_schema[i]);
             execute(query);
         }
+    } else {
+        // Limit history size to 2000 entries
+        QSqlQuery cleanupHistory = prepare(QString("DELETE FROM browser_history WHERE id NOT IN (SELECT id from "\
+                                                   "browser_history ORDER BY date DESC LIMIT %1);").arg(MAX_BROWSER_HISTORY_SIZE));
```
Comment (rojkov) [inline src/dbworker.cpp]: micro-optimization: a better way to extrapolate a string with a constant is to use the STR macro. This way the string would be extrapolated in build time, not run time.

Comment (siteshwar) [inline src/dbworker.cpp]: I thought about it but decided not to do it as it might not look very readable.

Comment (rojkov) [inline src/dbworker.cpp]: That's a valid point.

But I feel it's acceptable to go for a slight optimization as this SQL is simple enough to grasp even with some noise added.

**Comment by rojkov** [comment]
LGTM

**Pull request merged**